src/api/spec/services/meta_controller_service/project_updater_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.describe ::MetaControllerService::ProjectUpdater do
+  let(:project) { create(:project, name: 'openSUSE_41') }
+  let(:admin_user) { create(:admin_user, login: 'Admin') }
+
+  before do
+    User.current = admin_user
+  end
+
+  context 'with valid meta' do
+    let(:meta) do
+      <<-HEREDOC
+      <project name="#{project.name}">
+      <title/><description/>
+      </project>
+      HEREDOC
+    end
+
+    let(:project_updater) { ::MetaControllerService::ProjectUpdater.new(project: project, request_data: Xmlhash.parse(meta)) }
+
+    describe '#valid?' do
+      let(:project_updater_call) { project_updater.call }
+
+      it { expect(project_updater).to be_valid }
+    end
+
+    describe '#errors' do
+      let(:project_updater_call) { project_updater.call }
+
+      it { expect(project_updater.errors).to be_blank }
+    end
+  end
+
+  context 'with nonexistent path' do
+    let(:meta) do
+      <<-HEREDOC
+      <project name="#{project.name}">
+      <title/>
+      <description/>
+      <repository name="not-existent">
+      <path project="not-existent" repository="standard" />
+      </repository>
+      </project>
+      HEREDOC
+    end
+
+    let(:project_updater) { ::MetaControllerService::ProjectUpdater.new(project: project, request_data: Xmlhash.parse(meta)) }
+
+    describe '#valid?' do
+      let!(:project_updater_call) { project_updater.call }
+
+      it { expect(project_updater).not_to be_valid }
+    end
+
+    describe '#errors' do
+      let!(:project_updater_call) { project_updater.call }
+
```

Comment:
I would...
* remove the newline to have the `let`s grouped
* move `let!(:project_updater_call) { project_updater.call }` up to `context 'with nonexistent path'` since it's used in every `describe` block
* convert the `let` to a `before` block since it's just a method call. (IMO `subject` wouldn't be correct since we actually test `project_updater`, the object). It's just coincident that it returns the updater object. But I am splitting hairs, both `before` and `subject` would fit better than `let`.

That said, I give up:-)

Reply:
done @bgeuken. Please review